protovalidate/internal/extra_func.py

```diff
@@ -60,10 +63,37 @@ def validate_email(addr):
     return _validate_hostname(parts[1])
 
 
-def is_ip(val: celtypes.Value, version: typing.Optional[celtypes.Value] = None) -> celpy.Result:
-    if not isinstance(val, (celtypes.BytesType, celtypes.StringType)):
-        msg = "invalid argument, expected string or bytes"
-        raise celpy.EvalError(msg)
+def validate_host_and_port(string: str, *, port_required: bool) -> bool:
+    if string == "":
+        return False
+
+    split_idx = string.rfind(":")
+    if string[0] == "[":
+        end = string.find("]")
+        after_end = end + 1
+        if after_end == len(string):  # no port
+            return not port_required and validate_ip(string[1:end], 6)
+        if after_end == split_idx:  # port
+            return validate_ip(string[1:end]) and validate_port(string[split_idx + 1 :])
```

Comment:
```suggestion
            return validate_ip(string[1:end]) and validate_port(string[split_idx + 1:])
```

Not sure if we've got something configured to auto format the file but it might be worth looking into.